src/infrastructure/metrics.py

```
@@ -0,0 +1,229 @@
+
+    async def _report_loop(self):
+        while self._running:
+            try:
+                self._report_system_metrics()
+                self.metrics.flush()
+                await asyncio.sleep(self.interval)
+            except Exception as e:
+                logger.error(f"Error reporting metrics: {e}")
+
+    def _report_system_metrics(self):
+        import psutil
```

Pylint (3.3.7)

[convention] Import outside toplevel (psutil)

(C0415)